lib/features/family/features/topup/screens/topup_initial_bottom_sheet.dart

```diff
@@ -4,27 +4,34 @@ import 'package:font_awesome_flutter/font_awesome_flutter.dart';
 import 'package:givt_app/core/enums/country.dart';
 import 'package:givt_app/features/auth/cubit/auth_cubit.dart';
 import 'package:givt_app/features/family/features/admin_fee/presentation/widgets/admin_fee_text.dart';
+import 'package:givt_app/features/family/features/profiles/cubit/profiles_cubit.dart';
 import 'package:givt_app/features/family/features/topup/cubit/topup_cubit.dart';
 import 'package:givt_app/features/family/shared/widgets/content/amount_counter.dart';
+import 'package:givt_app/features/family/shared/widgets/inputs/input_checkbox.dart';
 import 'package:givt_app/features/family/shared/widgets/layout/givt_bottom_sheet.dart';
 import 'package:givt_app/shared/widgets/buttons/givt_elevated_button.dart';
 import 'package:givt_app/shared/widgets/common_icons.dart';
 import 'package:go_router/go_router.dart';
 import 'package:intl/intl.dart';
 
-class TopupInitialBottomSheet extends StatelessWidget {
+class TopupInitialBottomSheet extends StatefulWidget {
   const TopupInitialBottomSheet({
-    required this.topupAmount,
-    required this.amountChanged,
     super.key,
   });
 
-  final int topupAmount;
-  final void Function(int) amountChanged;
+  @override
+  State<TopupInitialBottomSheet> createState() =>
+      _TopupInitialBottomSheetState();
+}
```

Comment:
just out of curiosity, why was this necessary/ what does this do?

Reply:
I made it a stateful widget instead of stateless, because I was doing weird stuff first with a function to push back values. Thats now more properly handled by the cubit :)